common/ledger/dataformat/dataformats.go

```diff
@@ -9,16 +9,19 @@ package dataformat
 import "fmt"
 
 const (
-	// Version1x specifies the data format in version 1.x
-	Version1x = ""
+	// PreviousVersion specifies the data format in previous version (1.x for now)
+	PreviousVersion = ""
 
-	// Version20 specifies the data format in version 2.0
-	Version20 = "2.0"
+	// CurrentVersion specifies the data format in current version (2.0 for now)
+	CurrentVersion = "2.0"
 )
 
 var (
-	// Version1xBytes defines the []byte for Version1x to be nil
-	Version1xBytes []byte = nil
+	// PreviousVersionBytes specifies the []byte for PreviousVersion
+	PreviousVersionBytes []byte = nil
+
+	// CurrentVersionBytes specifies the []byte for CurrentVersion
+	CurrentVersionBytes = []byte(CurrentVersion)
```

Comment:
Do we need these? Can't the consumer simply convert the version string to bytes when needed?

Reply:
The format is stored as []byte in a db. There was no format key in fabric v1.x, so we compare to "nil" for upgrade eligibility. However, we cannot convert the PreviousVersion string to nil bytes. CurrentVersionBytes can be converted from CurrentVersion string, but it is added to be same as PreviousVersionBytes. I don't have strong opinion to keep or remove CurrentVersionBytes.

Reply:
A general suggestion is to avoid package-level variables (for Fabric). I am not getting into that. My question is more about why we need to define variables such as **XXXbytes** to just represent a byte version of **XXX** const. I am fine with this usage when it is used within a function as local variables but certainly not as exported variables (when both **XXX** and **XXXbytes** are exported). 

Can't we use 
```
if bytes.Equal(format, []byte(dataformat.PreviousVersion)) 
if bytes.Equal(format, []byte(dataformat.CurrentVersion))
```
 instead of 
```
if bytes.Equal(format, dataformat.PreviousVersionBytes)
if bytes.Equal(format, dataformat.CurrentVersionBytes)
```
?

Even if the stored version of the data is nil, can't we compare it against `[]byte(dataformat.PreviousVersion)` given that the `dataformat.PreviousVersion` is `""`? 

The following bytes.Equal would return true. 
```
var dbVersion1x []byte = nil
var previousVersion = ""
fmt.Println(bytes.Equal(dbVersion1x, []byte(previousVersion)))
fmt.Println(bytes.Equal(nil, []byte(previousVersion)))
fmt.Println(bytes.Equal(nil, []byte("")))
fmt.Println(bytes.Equal([]byte(nil), []byte("")))
```
You can play with this code here -- https://play.golang.org/p/WywdtwFTGvo

Let me know if I am missing anything.

Reply:
Thank you. Will remove XXXBytes.